Packs/IntSight/Integrations/Rapid7ThreatCommand/Rapid7ThreatCommand.py

```diff
@@ -0,0 +1,5354 @@
+    """
+    return reputation_handler(
+        args, client, file_reputation_handler, IOCType.FILE.value.lower()
+    )
+
+
+@polling_function(
+    name="ip",
+    interval=arg_to_number(demisto.args().get("interval_in_seconds", DEFAULT_INTERVAL)),
+    timeout=arg_to_number(demisto.args().get("timeout_in_seconds", DEFAULT_TIMEOUT)),
+    requires_polling_arg=False,
+)
+def ip_command(args: dict[str, Any], client: Client) -> PollResult:
+    """
+    Enrich ip IOC (Generic reputation command).
+
+    Args:
+        client (Client): Threat Command API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        PollResult: outputs, readable outputs and raw response for XSOAR.
+    """
+    return reputation_handler(
+        args, client, ip_reputation_handler, IOCType.IP.value.lower()
+    )
+
+
+@polling_function(
+    name="url",
+    interval=arg_to_number(demisto.args().get("interval_in_seconds", DEFAULT_INTERVAL)),
+    timeout=arg_to_number(demisto.args().get("timeout_in_seconds", DEFAULT_TIMEOUT)),
+    requires_polling_arg=False,
+)
+def url_command(args: dict[str, Any], client: Client) -> PollResult:
+    """
+    Enrich URL IOC (Generic reputation command).
+
+    Args:
+        client (Client): Threat Command API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        PollResult: outputs, readable outputs and raw response for XSOAR.
+    """
+
+    return reputation_handler(
+        args, client, url_reputation_handler, IOCType.URL.value.lower()
+    )
+
+
+@polling_function(
+    name="domain",
+    interval=arg_to_number(demisto.args().get("interval_in_seconds", DEFAULT_INTERVAL)),
+    timeout=arg_to_number(demisto.args().get("timeout_in_seconds", DEFAULT_TIMEOUT)),
+    requires_polling_arg=False,
+)
+def domain_command(args: dict[str, Any], client: Client) -> PollResult:
+    """
+    Enrich domain IOC (Generic reputation command).
+
+    Args:
+        client (Client): Threat Command API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        PollResult: outputs, readable outputs and raw response for XSOAR.
+    """
+    return reputation_handler(
+        args, client, domain_reputation_handler, IOCType.DOMAIN.value.lower()
+    )
+
+
+def reputation_handler(
+    args: dict[str, Any], client: Client, handler_command: Callable, key: str
+) -> PollResult:
+    """
+    Handle with all reputation commands.
+
+    Args:
+        client (Client): Threat Command API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+        handler_command (Callable): Handler command for each command.
+        key (str): Key for the IOC.
+
+    Returns:
+        PollResult: outputs, readable outputs and raw response for XSOAR.
+    """
+    execution_metrics = ExecutionMetrics()
+    execution_metrics.general_error += 1
+    ioc_values: List[str] = argToList(args[key])
+    current_file = ioc_values[0]
```

Comment:
not sure I understand why we take only the 1st one (and then remove it in the end)
from how I see it, in the first command execution we should run the enrich_ioc on the entire list, then poll for their statuses in the following executions.
let me know what am I missing

Reply:
Hi @dantavori, good point!
I implemented the `reputation_handler` command again, please review it :)